### ISSUE TITLE
Cannot write to keys named "path"

### ISSUE THREAD

**Issue by otakup0pe** [hvac/hvac#133]
Discovered this trying to query the [`sys/capabilities`](https://www.vaultproject.io/api/system/capabilities.html) endpoint, which takes both `path` and `token` as it's required arguments. 

We may wish to use something that isn't `**kwargs` in `Client.write`

```
In [9]: obj = {'path': 'sys/leader', 'token': client.token}

In [10]: client.write('sys/capabilities', **obj)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-10-b2eab5c7a994> in <module>()
----> 1 client.write('sys/capabilities', **obj)

TypeError: write() got multiple values for keyword argument 'path'
```

**Comment by dmyerscough** [comment]
@otakup0pe the error message is because you are using path twice e.g. `sys/capabilties` is a path and you have defined a `path` inside your obj dictionary.

You could do:-

```python
>>> client.write(**obj)
```

**Comment by otakup0pe** [comment]
Could kind of work around this by renaming the `path` argument on `write` but that seems like a kludge. Something else to think about as part of #95

Might do the kludge in the mean time, I'd love to be able to use the capabilities path...

**Comment by jeffwecan** [comment]
Thinking about this more lately, I figure there are two changes that would help see things resolved:

- Add explicit method(s) to cover the original source of the issue. I.e., add a method for "Query Token Capabilities"
- Change the write method to take a path argument and a params/payload/what-have-you argument instead of just using `**kwargs` generally.

**Comment by akdor1154** [comment]
@jeffwecan any chance of seeing this (particularly your second point) before 1.0? It would be a small but breaking change to most users of this library.

**Comment by jeffwecan** [comment]
Sure thing! I'll aim to get this in to the next minor release if possible (no guaranteed ETA along those lines at this point however I imagine it would come up over the next month or two).

**Comment by briantist** [comment]
> have this been resolved in the current release version? This milestone seems to be closed.

Hm, as far as I can tell this was never solved. This method is a generic "write" not tied to any specific backend. The current implementation has the same issue of it being unable to write a key with the name `path`.

I do think there is some utility in this method, and having it take a `data` argument (what Jeff called a "params/payload/what-have-you argument") would be a good way to workaround it.

To @akdor1154 's point, this would be a breaking change.

So I think what we could do here is a process of:
- add the "new" argument (`data` or whatever) in a minor release
- keep the current `kwargs`, but deprecate it
- eventually, in a future major release, remove `kwargs`

The only issue I see with this is that _any new argument added_ is yet another argument that will be "invalid" as supplied in `kwargs`, so for example as of today, while you could not supply a `path` key in `kwargs`, you _could_ supply a `data` key.

If we add a `data` argument, that will no longer be possible, Whatever name we choose, will be taken out of consideration for `kwargs`, which kind of breaks the idea of backwards compatibility, so adding it in a minor release may not be feasible.

We could look to do the adding (not the removing) in the `2.0.0` release, which is coming soon. My only concern with that is lack of notice as I'd prefer to have a decent amount of time to announce breaking changes before they are implemented, so we would have to either have this introduced on short notice (because `2.0.0` is going to be quite soon I think), or wait until `3.0.0` so that we can give notice.

In short: there's no perfect option here, I'm open to suggestions.

**Comment by kaypeter87** [comment]
Adding the reference here of the above PR so we can have some visibility with the issue.

https://github.com/hvac/hvac/pull/1028

**Comment by kaypeter87** [comment]
Related #138
